### ISSUE TITLE
F632: false positive

### ISSUE THREAD

**Issue by spaceone** [astral-sh/ruff#24887]
### Summary

```diff
$ ruff check --diff --select F632
-    assert Date() == [] is None  # noqa: E711
+    assert Date() == [] == None  # noqa: E711
```

### Version

0.15.10

**Comment by dylwil3** [comment]
Can you say a bit more? This looks like a true positive to me: `[] is None` compares `[]` to `None` using `is`, and the rule description says:

> Checks for is and is not comparisons against literals, like integers, strings, or lists.

Note that `Date() == [] is None` is parsed as a chained comparison so it is semantically equivalent to

```python
(Date() == []) and ([] is None)
```

and it is the term `[] is None` that is being flagged here.

**Comment by spaceone** [comment]
Oups, you are absolutely right. The (very old) code wasn't executed at all and wouldn't have passed, but I assumed it did and therefor didn't check the semantic carefully enough.
